### PR TITLE
Systemd nutshutdown sanity checks

### PR DESCRIPTION
**Pull request by jimklimov**
Follows up for #382 and #384

CC @bigon

### CODE REVIEW

**Comment by jimklimov** [comment]
No strong opinion here. Would 0 or 1s be good?

**Comment by bigon** [comment]
I would actually not modify nutshutdown at all

What are you trying to fix here?

**Comment by bigon** [comment]
We don't want to have POWEROFF_WAIT set by default, see https://github.com/networkupstools/nut/blob/master/docs/man/nut.conf.txt#L66

**Comment by jimklimov** [comment]
Fixing... I guess I was adding sanity-checks first for cases that only nut-server or nut-client is installed, so some binaries are not available. Then it struck me that we could have errors determining this delay and never check for results. Otherwise, same field as #382 / #384.

**Comment by jimklimov** [comment]
I believe the current implicit default, if systemd does not bail on lines
that errored (like `set -e`) if that nothing gets resolved, or a bad-number
token is resolved, and is passed as argument to `sleep` effecting a `0`...

**Comment by jimklimov** [comment]
Ok, I looked more at the script and think I missed `if [ -n "$wait_delay" ] ; then` line earlier, and that may be the point of your question :)

Can the sanity fallback be an empty value then, so only a present non-negative number triggers the wait-and-reboot?

**Comment by bigon** [comment]
The doc says that `POWEROFF_WAIT`is using a format compatible with `sleep(1)` that means that it's not an integer (could be 2m, 3h,...)

**Comment by jimklimov** [comment]
Depends on implementation of `sleep` but I see your point.

**Comment by jimklimov** [comment]
@bigon : is the current state of PR acceptably reasonable? :)

**Comment by jimklimov** [comment]
Note: as of now, only the sanity checks remain in the PR, no settings of defaults (beside emptying the delay variable in case of config parsing errors as decided by exit code).

I believe this is now safe(r) to merge.

File: scripts/systemd/nutshutdown.in [hunk 1]
```diff
@@ -1,7 +1,14 @@
 #!/bin/sh
 
+# This script requires both nut-server (drivers)
+# and nut-client (upsmon) to be present locally
+# and on mounted filesystems
+[ -x "@SBINDIR@/upsmon" ] && [ -x "@SBINDIR@/upsdrvctl" ] || exit
```
[the remark this answers is not in the document]
Comment (zykh) [reply scripts/systemd/nutshutdown.in]: like the `POWERDOWNFLAG` file?

**Pull request merged**